fe/fe-core/src/main/java/com/starrocks/catalog/RecyclePartitionInfo.java

```diff
@@ -38,6 +38,12 @@ public abstract class RecyclePartitionInfo extends JsonWriter {
     @SerializedName(value = "recoverable")
     protected boolean recoverable;
 
+    /**
+     * partition retention period in `second` unit
+     */
+    @SerializedName(value = "retentionPeriod")
+    protected long retentionPeriod = 0L;
```

Comment:
may need to add one line in the default constructor, in case it is recovered by Gson deserializer.